devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -609,6 +612,15 @@ bool YarpRobotLoggerDevice::attachAll(const yarp::dev::PolyDriverList& poly)
 {
     constexpr auto logPrefix = "[YarpRobotLoggerDevice::attachAll]";
 
+    bool ok = true;
+    // open the TextLogging port
+    ok = ok && m_textLoggingPort.open(m_textLoggingPortName);
+    // run the thread
+    m_lookForNewLogsThread = std::thread([this] { this->lookForNewLogs(); });
```

Comment:
I think we miss a mutex for the ``m_textLoggingPortNames``

Reply:
Fixed in [5f62eea](https://github.com/ami-iit/bipedal-locomotion-framework/commit/5f62eeab579849a554b67ead618f1b1f6909518c)